src/squishy/core.clj

```diff
@@ -61,6 +61,21 @@
   `(with-backoff* ~backoff-start ~max-failures ~consumer-id
      (fn [~reset] ~@body)))
 
+(defn pre-deleting-consumer
+  "Setting :no-retries to true will use this pre-deleting consumer which
+   deletes the message before processing. Generally this is a Bad Idea (TM),
+   but if you have to sometimes process messages that take longer than the
+   maximum visibility timeout, you have little choice but to delete the message
+   and hope processing completes normally."
```

Comment:
So I'm fairly clueless about how SQS works, but are you saying that there's a possibility that by deleting ahead of time we'll lose the message and processing won't actually happen?

Reply:
Yes, there will be this chance with this solution, which is why I'm not super thrilled with it. Thinking about it now, maybe we could offer a version that provided a callback function that, when called, would delete the message. This would allow, for instance, `data-processor` to call the callback at a point where it has kicked off processing and has recorded enough information in the db. This would make sure nothing ever got lost between receiving the message and recording enough information that we've started the job. Thoughts?

Reply:
That seems smart to me, especially if the issues with this have more to do with the code using this library vs. this library--again, not knowing much about SQS, maybe a question to ask is, which solution will be more useful longer-term to users of this library?

Reply:
I think my proposed second solution is maybe better overall. It's really only relevant to users who may have processes so long that they go past the 12 hour mark, and that's hopefully vanishingly small. But either solution I proposed here will require alterations in the codebases using the library to switch over to this functionality, so I think my second proposal is better overall, and in fact if they never call the callback, it'll act like the original solution I think.

Reply:
Let me know if you want me to re-review after tweaking it, or approve as is...